packages/frontend/src/components/nft/NFTDetailWrapper.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+
+import { selectAccountId, selectBalance } from "../../redux/slices/account";
+import NonFungibleTokens from "../../services/NonFungibleTokens";
+import { NFTDetail } from "./NFTDetail";
+
+export function NFTDetailWrapper ({
+    match,
+    history
+}) {
+    const accountId = useSelector(selectAccountId);
+    const balance = useSelector(state => selectBalance(state));
+    const nearBalance = balance.balanceAvailable;
+
+    const contractId = match.params.contractId;
+    const tokenId = match.params.tokenId;
+    const [ nft, setNft ] = useState();
+    const [ ownerId, setOwnerId ] = useState();
+
+    useEffect(() => {
```

Comment:
1. This data should be loaded from Redux  -- it will usually already be there, since if someone is transferring an NFT they have likely already viewed it in the wallet UI and clicked on the NFT to get NFT details, and the wallet UI is already populated using Redux data.

2. If the NFT contract and owned token metadatas aren't already in Redux, we should be dispatching actions to load those into redux here, rather than only loading it into this component's local `useState()` hooks.  

3. This also means that we can get rid of both the `setState` calls for `ownerId` and for `nft`; both of these pieces of data would be loaded directly from Redux using selectors.